src/consolidations/main.eas

```diff
@@ -227,7 +235,9 @@ accum_loop:
   ;; Precompute record_offset = i*RECORD_SIZE.
   dup1                  ;; [i, i, count, head_idx, tail_idx]
   push RECORD_SIZE      ;; [size, i, i, count, head_idx, tail_idx]
-  mul                   ;; [record_offset, i, count, head_idx, tail_idx]
+  mul                   ;; [offset, i, count, head_idx, tail_idx]
+  push 1                ;; [1, offset, i, count, head_idx, tail_idx]
+  add                   ;; [record_offset, i, count, head_idx, tail_idx]
```

Comment:
I wonder if it is better to add a loop-level item `record_offset` and just add `RECORD_SIZE` during each iteration. Feels weird to `+1` every calculation.

Reply:
Yeah, so that would work as well. Let me know if you prefer this.

Reply:
I even thought turning the whole thing around to keep the offset permanently near the stack top and increment after every store. But then we'd need to find a clean way to hoist up the items from below.

Reply:
Eh looking into this more, this is pretty straightforward and understandable implementation. I think we can merge as-is and decide later if it makes sense to optimize it.

Reply:
It's less about optimizing and more about correctness. The advantage of a rolling offset is that we only need to add the length of the current item after writing the item. It removes the hard-coded offsets. But I can look into this as a refactoring after this PR is in.